graylog2-web-interface/src/stores/streams/StreamRulesStore.ts

```diff
@@ -75,6 +76,9 @@ class StreamRulesStore {
   _emitChange() {
     this.callbacks.forEach((callback) => callback());
   }
+  unregister(callback: Callback) {
+    lodash.pull(this.callbacks, callback);
```

Comment:
The fact that it is not an explicit dependency yet is a problem, agreed. We use it in another [place](https://github.com/Graylog2/graylog2-server/blob/master/graylog2-web-interface/src/stores/alertconditions/AlertConditionsStore.js#L77) already, so this change is not introducing it in the first place and I think both future and existing code could benefit from it, once we are using it. Documentation is excellent too, so there is no real barrier for people in the team not knowing it yet.